samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c

```diff
@@ -197,6 +200,24 @@ static int cloud_lwm2m_init(const struct device *unused)
 	return 0;
 }
 
+static void cloud_lwm2m_rd_client_update_lifetime(int srv_obj_inst)
+{
+	char pathstr[MAX_RESOURCE_LEN];
+	uint32_t current_lifetime = 0;
+	uint32_t lifetime = CONFIG_LWM2M_ENGINE_DEFAULT_LIFETIME;
+
+	snprintk(pathstr, sizeof(pathstr), "1/%d/1", srv_obj_inst);
+	lwm2m_engine_get_u32(pathstr, &current_lifetime);
+	if (current_lifetime != lifetime) {
```

Comment:
I guess we shortly discussed yesterday whether this check is needed. Is it? What happens if we remove it?

Reply:
@jheiskan81 Can you answer this? Basically this comes down to what happens if `lwm2m_engine_set_u32()` is called with the same value? If `lwm2m_engine_set_u32()` could be called always, we could remove the `update_session_lifetime` flag and the `current_lifetime != lifetime` check, simplifying the implementation quite a bit.

Looking at the code in `lwm2m_engine_set_u32()` and `lwm2m_engine_set()`, it does check if the value has changed before calling `lwm2m_notify_observer_path()`. I guess setting the same value would not cause any data traffic then? Setting of the session lifetime is triggered by `LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE` and `LWM2M_RD_CLIENT_EVENT_REG_UPDATE_COMPLETE` events, so it would not happen often with `modem_shell`, where the session lifetime is 12 hours and registration update is configured to be sent 10 minutes before session expiration.

Reply:
When client do bootstrap server may define new lifetime which could shorter what we like to use. After LwM2M registration an connecting app verfify is lifetime shorter than wanted. By updating lifetime will generate Registration update when lifetime is changed.  This probably could be handled inside lwm2m engine but let use this until engine can handle this inside. @SeppoTakalo Shuold we handle this lifetime update inside lwm2m engine it probably nice way to do?

Reply:
`CONFIG_LWM2M_ENGINE_DEFAULT_LIFETIME` sets the default lifetime. I think in real use cases, it's desirable to be able to change the lifetime from the server side. So, always overwriting the session lifetime with the UE default value might not be the best idea, although for `modem_shell` it would be probably fine.

Should this be done only after bootstrapping? The implementation is now copied from the LwM2M client sample. Should the `update_session_lifetime` flag be set to true only after bootstrapping has been performed? And in that case the `update_session_lifetime` should be probably initialized to `false` instead of `true` (like it is now).

Reply:
Pushed a simplified version, @trantanen please have a look. Based on my tests this works. Session lifetime can also be changed from the server side and it doesn't get overwritten by the default value.